### PR TITLE
Moving flood control mechanism to TfaLoginForm / EntryForm

### PR DESCRIPTION
**Pull request by vg3095**
Since Tfa / TfaManager is to be deprecated , I have moved flood control related functions to relevant TfaLoginForm / EntryForm classes.
Please Review.

### CODE REVIEW

File: config/install/tfa.settings.yml [hunk 1]
```diff
@@ -3,4 +3,9 @@ setup_plugins: { }
 send_plugins: { }
 login_plugins: { }
 validate_plugins: { }
-validate_weights: { }
\ No newline at end of file
+validate_weights: { }
+tfa_user_window: 900
+tfa_test_mode: 0
+tfa_flood_window: 3600
+tfa_user_threshold: 6
```
Comment (nerdstein) [inline config/install/tfa.settings.yml:10]: Just to confirm, this is a global setting and defined for each user?

File: src/Form/EntryForm.php [hunk 2]
```diff
@@ -51,13 +59,14 @@ public function getFormID() {
    */
   public function buildForm(array $form, FormStateInterface $form_state, AccountInterface $user = null) {
     // Check flood tables.
-    //@TODO Reimplement Flood Controls.
+
```
Comment (nerdstein) [inline src/Form/EntryForm.php:60]: I'm curious why you didn't implement the flood controls here -- they are commented out.

Comment (nerdstein) [inline src/Form/EntryForm.php:60]: Please note - I'm just asking why, not suggesting they should be. If you implemented flood controls, maybe we should consider removing the commented out code

File: src/Form/EntryForm.php [hunk 3]
```diff
@@ -154,6 +163,68 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
...
+    return TRUE;
+  }
+
+  /**
+   * Check if flood has been hit.
+   * @param Tfa $tfa
+   * @return bool
+   */
+   // Argument of this function has to be changed , depending upon getContext function
+  public function _tfa_hit_flood($tfa) {
```
Comment (nerdstein) [inline src/Form/EntryForm.php]: Is this a public function? The underscore in the function name suggests its likely private and you do not need the underscore at the beginning of the function name.

Proper class naming conventions for methods would be tfaHitFloor

File: tfa.services.yml [hunk 4]
```diff
@@ -2,6 +2,9 @@ services:
   plugin.manager.tfa.login:
     class: Drupal\tfa\TfaLoginPluginManager
     parent: default_plugin_manager
+  plugin.manager.tfa.base:
```
Comment (nerdstein) [inline tfa.services.yml]: Hmm... I don't think 'base' is a plugin manager, is it? Should this be here?

I think this should be removed

File: config/install/tfa.settings.yml (hunk 1 above)
Comment (vg3095) [inline config/install/tfa.settings.yml:10]: Yes, these are global settings and defined for each user.

File: src/Form/EntryForm.php (hunk 2 above)
Comment (vg3095) [inline src/Form/EntryForm.php:60]: _tfa_hit_flood hook needs an argument , through which we can access getContext() function.
At this point, I don`t know where  getContext() function will be moved (presently ,it is in Tfa class , which is deprecated).

File: src/Form/EntryForm.php (hunk 3 above)
Comment (vg3095) [inline src/Form/EntryForm.php]: My bad , I din`t know that , will update.

File: tfa.services.yml (hunk 4 above)
Comment (vg3095) [inline tfa.services.yml]: Ok , I removed it.

**Comment by vg3095** [comment]
Updated with some changes , please review.

**Pull request closed without merging**